src/base/bittorrent/sessionimpl.cpp

```diff
@@ -4898,14 +4899,15 @@ void SessionImpl::handleTorrentMetadataReceived(TorrentImpl *const torrent)
 
 void SessionImpl::handleTorrentPaused(TorrentImpl *const torrent)
 {
-    torrent->resetTrackerEntries();
+    torrent->resetTrackerEntryInfo();
 
-    const auto &trackerEntries = torrent->trackers();
-    QHash<QString, TrackerEntry> updatedTrackerEntries;
-    updatedTrackerEntries.reserve(trackerEntries.size());
-    for (const auto &trackerEntry : trackerEntries)
-        updatedTrackerEntries.emplace(trackerEntry.url, trackerEntry);
-    emit trackerEntriesUpdated(torrent, updatedTrackerEntries);
+    const QVector<TrackerEntryInfo> trackerInfo = torrent->trackers();
```

Comment:
The noun info is uncountable. The plural form of info is also info.

Reply:
> The noun info is uncountable. The plural form of info is also info.

We are not talking about classical English. For code readability, it is important to distinguish between the name of a single item and the name of a set (list, array etc.) of such items.

BTW, "infos" is plural of "info" according to "Webster's New International English Dictionary. 2012".

Reply:
>We are not talking about classical English. 

It has nothing to do with classical English. 'Uncountable' is what a modern English dictionary would tell you and 99% of google search will tell you the same.
https://dictionary.cambridge.org/us/grammar/british-grammar/information
https://www.collinsdictionary.com/dictionary/english/information
https://www.oxfordlearnersdictionaries.com/definition/english/info?q=info

>BTW, "infos" is plural of "info" according to "Webster's New International English Dictionary. 2012".

Do you mind giving links to back you up? I couldn't find it.

>For code readability, it is important to distinguish between the name of a single item and the name of a set (list, array etc.) of such items.

I just want to clarify that you are intentionally not following the rules. This might trigger/confuse a native English speaker when reviewing/maintaining the code.

For the code here, we can use alternatives such as `trackerInfoList` or `trackers`, do you object?

Reply:
> I just want to clarify that you are intentionally not following the rules. This might trigger/confuse a native English speaker when reviewing/maintaining the code.

Personally, I try to avoid using words like "info", "data" etc. in such cases where they can lead to ambiguous names (if used strictly according to the rules of English).
On the other hand, we really do not use a purely English language, but some surrogate based on it, so we can freely interpret TrackerEntryInfo as a separate term that is countable.

> For the code here, we can use alternatives such as `trackerInfoList` or `trackers`, do you object?

Of course, I don't mind any alternatives that will make it easy to distinguish a single item from a collection.
However, I would not like it to be reduced to adding a type name in each case, such as trackerEntryInfoMap, trackerEntryInfoHash etc. (the suffix List can be used without referring to a specific type).
BTW, using the previous names would also be quite acceptable to me.



> > BTW, "infos" is plural of "info" according to "Webster's New International English Dictionary. 2012".
> 
> Do you mind giving links to back you up? I couldn't find it.

I just came across an indirect referring to it when I was looking for "infos". Anyway, it's not as important to me as the reasoning above.

Reply:
Perhaps it would be better not to rename anything at all, but to create a new structure with a new name, e.g., `TrackerEntryDescritor`.

Reply:
>Perhaps it would be better not to rename anything at all, but to create a new structure with a new name, e.g., TrackerEntryDescriptor.

I don't get what you mean. Does that imply you are going to handle it yourself?

Reply:
> > Perhaps it would be better not to rename anything at all, but to create a new structure with a new name, e.g., TrackerEntryDescriptor.
> 
> I don't get what you mean.

I mean to go the other way (instead of what was originally suggested in this PR):
1. Do not rename the existing `TrackerEntry` structure,
2. Use a new name for the new structure, e.g., `TrackerEntryDescriptor`. 

The pros are:
1. This will affect only the code that requires changes (i.e. the one where only the URL and tier are needed without additional fields from original TrackerEntry),
2. We will avoid using these ambiguous names based on "info".

Reply:
> I consider this PR still has its value

I'm not arguing with that. The question is, what is the main goal? If "create a simple struct to pass/parse information from the GUI (without all the additional unused fields as it is now currently)" is what it is (in my opinion it should be exactly that), then as I pointed out above, this value can be achieved in a much simpler way without affecting so much code.

Reply:
>then as I pointed out above, this value can be achieved in a much simpler way without affecting so much code.

IMO it would not be significantly simpler, maybe just a tiny bit. This change still affects a non-trivial amount of code.

>Or do you still insist that renaming TrackerEntry to TrackerEntryInfo is important in itself?

Yes. The current `TrackerEntry` is badly named. Someone would do the cleanup sooner or later.

Reply:
> The current `TrackerEntry` is badly named.

It looks subjective. However, I don't mind seeing it as a `TrackerEntryInfo`, if we can avoid the ambiguity because of having `Info` in it.

Reply:
>However, I don't mind seeing it as a TrackerEntryInfo, if we can avoid the ambiguity because of having Info in it.

Not sure what ambiguity you are talking about except the downside of 'info' being an uncountable noun.

Another idea, perhaps statistics/stats can be used instead of info.

Reply:
> Not sure what ambiguity you are talking about except the downside of 'info' being an uncountable noun.

Yes, that's exactly what it's about. Therefore, we will have to avoid naming variables simply by analogy with the class name, as we could do with other names, e.g. `TrackerEntry trackerEntry` and `QList<TrackerEntry> trackerEntries`.
Of cource there should not be a problem to use something like `TrackerEntryInfo tracker` and `QList<TrackerEntryInfo> trackers` in local scopes. We can even use the "name by analogy with the class name" (i.e. `trackerEntryInfo`) for single objects, and avoid them only for collections.

> Another idea, perhaps statistics/stats can be used instead of info.

Statistics/stats have the same disadvantage like "info".

Reply:
I will continue reviewing.

Reply:
>Statistics/stats have the same disadvantage like "info".

What disadvantage? Statistics/stats is countable unlike info.

Reply:
> Statistics/stats is countable unlike info.

Well, it looks like it is. Maybe it's my poor English... But nevertheless, I do not recall that "statistic" is widely used. Usually only "statistics" is found.
In addition, "statistics" is hardly an appropriate term to talk about "current indicators". IMO, "status" would be more suitable, but we already use it in a different (narrower) sense in relation to tracker entries.

Reply:
> IMO, "status" would be more suitable, but we already use it in a different (narrower) sense in relation to tracker entries.

So maybe rename current tracker-related "status" terms (e.g. `TrackerEndpointStatus`) to "state" (e.g. `TrackerEndpointState`) and `TrackerEntryInfo` to `TrackerEntryStatus`?

Reply:
>So maybe rename current tracker-related "status" terms (e.g. TrackerEndpointStatus) to "state" (e.g. TrackerEndpointState) and TrackerEntryInfo to TrackerEntryStatus?

Done.